src/CsvReader.php

```diff
@@ -62,7 +62,11 @@ public function __construct(
         $this->validateLineBreak();
 
         rewind($this->filePointer);
-        $this->header = UTF8BOMHelper::detectAndRemoveBOM($this->readLine());
+        $header = UTF8BOMHelper::detectAndRemoveBOM($this->readLine());
+        if (is_array($header) && $header[0] === null) {
```

Comment:
Nakonec místo exception ukládám do headeru prázdné pole ať se to chová stejně, jako když je to CSV úplně prázdné (`\Keboola\Csv\Tests\CsvReadTest::testEmptyCsv `)